packages/x-data-grid/src/hooks/features/rows/useGridParamsApi.ts

```diff
@@ -50,7 +55,10 @@ export function useGridParamsApi(apiRef: React.MutableRefObject<GridPrivateApiCo
 
   const getCellParams = React.useCallback<GridParamsApi['getCellParams']>(
     (id, field) => {
-      const colDef = apiRef.current.getColumn(field);
+      const getColumnFn = props.unstable_listView
+        ? apiRef.current.getListColumn
+        : apiRef.current.getColumn;
+      const colDef = getColumnFn(field) as GridStateColDef;
```

Comment:
I would also consider abstracting the `if (listView) then getListColumn else getColumn` logic in a single function (likely `getColumn`), it avoids introducing a check on `listView` in unrelated places such as this one.

Also I wonder if there would be a way to set the visible columns in a way that avoids adding diverging functions for list view and normal view.

Reply:
Agreed. I thought it may be a good idea to have list view state and API methods completely separate from columns to start with, but doing this check in multiple places is not ideal.

I have added this to a follow up issue for implementation improvements https://github.com/mui/mui-x/issues/14997.

Reply:
I think setting up the state/model in a way that avoids those checks would be preferable to do in the initial PR (unless we have a very tight deadline, not sure what's the situation), and this comment is linked to [that one](https://github.com/mui/mui-x/pull/14393#discussion_r1802607256).

Reply:
I am hesitant to change it now as [we plan to release this tomorrow](https://www.notion.so/x-grid-Mobile-view-for-Data-Grid-b6228123c8a64352ba2c355c46e4e4ff?d=112cbfe7b66080f0ba5f001cfc35bf8d&pvs=4#5c1fee4d0ec04e629ddf32270f70a1e8). I will prioritise this post-merge.